docs/admin/authentication.md

```diff
@@ -60,19 +59,213 @@ more secure modes described above easier to use.
+- name: bob-the-bot-token-1yvwg
+```
+
+The created secret holds the public CA of the apiserver and a signed JSON Web
+Token (JWT).
+
+```
+$ kubectl get secret bob-the-bot-token-1yvwg -o yaml
+apiVersion: v1
+data:
+  ca.crt: (APISERVER'S CA BASE64 ENCODED)
+  token: (BEARER TOKEN BASE64 ENCODED)
+kind: Secret
+metadata:
+  # ...
+type: kubernetes.io/service-account-token
+```
+
+Note: values are base64 encoded because secrets are always base64 encoded.
+
+The signed JWT can be used as a bearer token to authenticate as the given service
+account. Normally these secrets are mounted into pods for in-cluster access to
+the apiserver, but can be used from outside the cluster as well.
+
+Service accounts authenticate with the username `system:serviceaccount:(NAMESPACE):(SERVICEACCOUNT)`,
+and are assigned to the groups `system:serviceaccounts` and `system:serviceaccounts:(NAMESPACE)`.
+
+WARNING: Because service account tokens are stored in secrets, any user with
+read access to those secrets can authenticate as the service account. Be cautious
+when granting permissions to service accounts and read capabilities for secrets.
+
+## OpenID Connect Tokens
+
+[OpenID Connect](https://openid.net/connect/) is a flavor of OAuth2 supported by
+some OAuth2 providers, notably Azure Active Directory, Salesforce, and Google.
+The protocol's main extension of OAuth2 is an additional field returned with
+the access token called an [ID Token](https://openid.net/specs/openid-connect-core-1_0.html#IDToken).
+This token is a JSON Web Token (JWT) with well known fields, such as a user's
+email, signed by the server. When used as a bearer token, the apiserver can
+verify ID token's signature and determine the end users identity.
+
+To enable the plugin, pass the following required flags:
+
+* `--oidc-issuer-url` URL of the provider which allows the apiserver to discover
+public signing keys. Only URLs which use the `https://` scheme are accepted.
+* `--oidc-client-id` A client id that all tokens must be issued for.
+
+Importantly, the apiserver is not an OAuth2 client, rather it can only be
+configured to trust a single client. This allows the use of public providers,
+such as Google, without trusting credentials issued to third parties. Admins who
+wish utilize multiple OAuth clients should explore providers which support the
+`azp` (authorized party) claim, a mechanism for allowing one client to issue
+tokens on behalf of another.
+
+The plugin also accepts the following optional flags:
+
+* `--oidc-ca-file` Used by the apiserver to establish and verify the secure connection to the issuer.
+* `--oidc-username-claim` JWT claim to use as the user name. By default `sub`,
+which is expected to be a unique identifier of the end user. Admins can choose
+other claims, such as `email`, depending on their provider.
+* `--oidc-groups-claim` JWT claim to use as the user's group. If the claim is present
+it must be an array of strings.
+
+## Webhook Token Authentication
```

Comment:
Have you actually wired this up?  It was my intent with https://github.com/kubernetes/kubernetes/pull/29796, but I've only actually wired it in openshift for node authentication.